src/search/FindBar.js

```diff
@@ -324,7 +339,9 @@ define(function (require, exports, module) {
                             searchHistory.pop();
                         }
                     }
-                    searchHistory.unshift($('#find-what').val());
+                    if ($('#find-what').val()) {
+                        searchHistory.unshift($('#find-what').val());
```

Comment:
I second @petetnt. Value of `#find-what` should be assigned to a variable while initiating the search session. Whole session should refer to the same value for navigation.